src/daemon/ca/status.rs

```diff
@@ -65,49 +68,130 @@ impl Default for CaStatus {
 
 pub struct StatusStore {
     store: KeyValueStore,
-    locks: Arc<CaLocks>,
+    cache: RwLock<HashMap<Handle, Arc<CaStatus>>>,
 }
 
 impl StatusStore {
     pub fn new(work_dir: &Path, namespace: &str) -> KrillResult<Self> {
         let store = KeyValueStore::disk(work_dir, namespace)?;
+        let cache = RwLock::new(HashMap::new());
 
-        // Create the per-CA lock structure so that we can guarantee safe access to each CA, while allowing
-        // multiple CAs in a single Krill instance to interact: e.g. a child can talk to its parent and they
-        // are locked individually.
-        let locks = Arc::new(CaLocks::default());
-        Ok(StatusStore { store, locks })
+        let store = StatusStore { store, cache };
+        store.warm()?;
+
+        Ok(store)
     }
 
-    fn status_key(ca: &Handle) -> KeyStoreKey {
-        KeyStoreKey::scoped(ca.to_string(), "status.json".to_string())
+    /// Load existing status from disk, support the pre 0.9.5 format and silently
+    /// convert it if needed.
+    fn warm(&self) -> KrillResult<()> {
+        for scope in self.store.scopes()? {
+            if let Ok(ca) = Handle::from_str(&scope) {
+                self.convert_pre_0_9_5_full_status_if_present(&ca)?;
+                self.load_full_status(&ca)?;
+            }
+        }
+
+        Ok(())
     }
 
-    /// Returns the stored CaStatus for a CA, or a default (empty) status if it can't be found
-    pub async fn get_ca_status(&self, ca: &Handle) -> Arc<CaStatus> {
-        // Try to get it from disk, if it's missing or corrupt - get a default
-        let lock = self.locks.ca(ca).await;
-        let _ = lock.read().await;
+    /// Load current status from disk, to be used when starting up. Iif there are any
+    /// issues parsing data then default values are used - this data is not critical
```

Comment:
Is this statement about this functionality not being critical compatible with the use of `unwrap()` calls in the fn below? Could it panic and cause Krill to exit while Krill could actually have happily continued running?

Reply:
can't panic, but pushed a fix to avoid calling `unwrap()` - should be better and less confusing